Sources/Core/Views/GenericButton/GenericButton.swift

```diff
@@ -0,0 +1,176 @@
+//
+//  GenericButton.swift
+//  PovioKit
+//
+//  Created by Arlind Dushi on 3/17/22.
+//  Copyright © 2022 Povio Inc. All rights reserved.
+//
+
+import SwiftUI
+
+@available(iOS 13, *)
+public class GenericButtonProperties: ObservableObject {
+  @Published public var title: String = ""
+  @Published public var font: Font = .system(size: 18)
+  @Published public var textColor: Color = .white
+  @Published public var cornerRadius: GenericButton.CornerRadiusType = .custom(0)
+  @Published public var backgroundType = GenericButton.Background.plain(.blue)
+  @Published public var borderColor: Color = .clear
+  @Published public var borderWidth: CGFloat = 0
+  @Published public var extraImage: GenericButton.ExtraImage?
+}
+
+@available(iOS 13, *)
+public struct GenericButton: View {
+  @ObservedObject public var properties = GenericButtonProperties()
+  private var actionHandler: (() -> Void)?
+  
+  public init() {}
+  
+  public enum Background {
+    case plain(Color)
+    case linearGradient(LinearGradient)
+    case radialGradient(RadialGradient)
+    case angularGradient(AngularGradient)
+  }
+  
+  public var body: some View {
+    GeometryReader { geo in
+      ZStack(alignment: getAlignment(for: properties.extraImage)) {
+        ButtonView(action: buttonAction, properties: properties)
+        ExtraImageView(imageAlignment: properties.extraImage, tintColor: properties.textColor, size: geo.size)
+      }
+    }
+  }
+  
+  public mutating func setAction(action: @escaping () -> Void) {
+    actionHandler = action
+  }
+}
+
+// MARK: - Public Properties
+@available(iOS 13, *)
+public extension GenericButton {
+  enum ExtraImage {
+    case left(Image)
+    case center(Image)
+    case right(Image)
+  }
+  
+  enum CornerRadiusType {
+    case rounded
+    case custom(CGFloat)
+  }
+}
+
+// MARK: - Private Methods
+@available(iOS 13, *)
+private extension GenericButton {
+  func buttonAction() {
+    actionHandler?()
+  }
+  
+  func getAlignment(for extraImage: ExtraImage?) -> Alignment {
+    switch extraImage {
+    case .some(let alignment):
+      switch alignment {
+      case .right:
+        return .trailing
+      case .left:
+        return .leading
+      case .center:
+        return .center
+      }
+    case .none:
+      return .trailing
+    }
+  }
+}
+
+// MARK: Views
+@available(iOS 13, *)
+private extension GenericButton {
+  struct ButtonView: View {
+    var action: () -> Void
+    var properties: GenericButtonProperties
+    
+    var body: some View {
+      GeometryReader { geo in
```

Comment:
Maybe it is better to avoid GeometryReader as much possible since it does not have predictable results: [[GeometryReader Discrepancies with Previous OS Versions](https://swiftui-lab.com/geometryreader-bug/)](https://swiftui-lab.com/geometryreader-bug/)
In this case, I think we can pass the size from parent view above:
`ButtonView(size: geo.size, action: buttonAction, properties: properties)`
and use it here instead of the GeometryReader:
```
private extension GenericButton {
  struct ButtonView: View {
    var size: CGSize
    var action: () -> Void
    var properties: GenericButtonProperties
    ...
```